ts/navigation/AuthenticationNavigator.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from "react";
+import { createStackNavigator } from "@react-navigation/stack";
+import CardSelectionScreen from "../screens/authentication/CardSelectionScreen";
+import CieAuthorizeDataUsageScreen from "../screens/authentication/cie/CieAuthorizeDataUsageScreen";
+import CieCardReaderScreen from "../screens/authentication/cie/CieCardReaderScreen";
+import CieConsentDataUsageScreen from "../screens/authentication/cie/CieConsentDataUsageScreen";
+import CieExpiredOrInvalidScreen from "../screens/authentication/cie/CieExpiredOrInvalidScreen";
+import CiePinLockedTemporarilyScreen from "../screens/authentication/cie/CiePinLockedTemporarilyScreen";
+import CiePinScreen from "../screens/authentication/cie/CiePinScreen";
+import CieWrongCiePinScreen from "../screens/authentication/cie/CieWrongCiePinScreen";
+import IdpLoginScreen from "../screens/authentication/IdpLoginScreen";
+import IdpSelectionScreen from "../screens/authentication/IdpSelectionScreen";
+import LandingScreen from "../screens/authentication/LandingScreen";
+import SpidCIEInformationScreen from "../screens/authentication/SpidCIEInformationScreen";
+import SpidInformationScreen from "../screens/authentication/SpidInformationScreen";
+import TestAuthenticationScreen from "../screens/authentication/TestAuthenticationScreen";
+import MarkdownScreen from "../screens/development/MarkdownScreen";
+import { AuthenticationParamsList } from "./params/AuthenticationParamsList";
+import ROUTES from "./routes";
+
+const Stack = createStackNavigator<AuthenticationParamsList>();
+
+const AuthenticationStackNavigator = () => (
+  <Stack.Navigator
+    initialRouteName={ROUTES.AUTHENTICATION_LANDING}
+    headerMode={"none"}
+    screenOptions={{ gestureEnabled: true }}
+  >
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_LANDING}
+      component={LandingScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_IDP_SELECTION}
+      component={IdpSelectionScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_CIE}
+      component={CardSelectionScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_IDP_LOGIN}
+      component={IdpLoginScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_IDP_TEST}
+      component={TestAuthenticationScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_SPID_INFORMATION}
+      component={SpidInformationScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_SPID_CIE_INFORMATION}
+      component={SpidCIEInformationScreen}
+    />
+
+    <Stack.Screen name={ROUTES.MARKDOWN} component={MarkdownScreen} />
+
+    <Stack.Screen
+      name={ROUTES.CIE_EXPIRED_SCREEN}
+      component={CieExpiredOrInvalidScreen}
+    />
+
+    <Stack.Screen name={ROUTES.CIE_PIN_SCREEN} component={CiePinScreen} />
```

Comment:
The same of the previous comments, also for `CiePinScreen`. In addition, in this screen seems that the navigation prop is never used, maybe can be removed?
In this screen also we have the usage of the hook `useNavigation`, can be worth adding the typing, in order to have typesafe navigation?

Reply:
@fabriziofff I have removed navigation props. What you mean with `can be worth adding the typing, in order to have typesafe navigation`?

Reply:
If we add the typing to the hook `useNavigator`, all the navigation actions will be typesafe too. You can see https://docs.google.com/document/d/1Bme0eHFIe431eR0Qc4KTiDrmjjW4hTsj0U_aXGcKjDA/edit#heading=h.ud0nmm2wsnx for further details.